### PR TITLE
Workflow time decrease

### LINKED ISSUE #16: Reduce workflow runtime
I am currently working on caching some major directories that could save 10-20 minutes per run. This is currently being tested on my branch and I will let you know what I discover.

### PR DESCRIPTION
**Pull request by ahadley1124**
Adding commit from #16 and @Okkonen-GitHub's previous changes to main

### CODE REVIEW

**Comment by sshine** [comment]
I'd like if we can merge @Okkonen-GitHub's changes as a separate PR to #17.

I can do the paperwork.

**Comment by ahadley1124** [comment]
@sshine That would be awesome. Thanks!

**Comment by Okkonen-GitHub** [comment]
Sure go ahead, I'll skip reviewing this them and wating for the sepate PRs.

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "icons/icon.icns",
         "icons/icon.ico"
       ],
-      "identifier": "com.tauri.dev",
+      "identifier": "us.austinh.simplex-desktop",
```
Comment (sshine) [inline src-tauri/tauri.conf.json:32]: Is this package identifier similar to Java namespaces?

I'm not sure what function it provides, and what changing it does.

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -26,3 +27,6 @@ default = [ "custom-protocol" ]
 # this feature is used for production builds where `devPath` points to the filesystem
 # DO NOT remove this
 custom-protocol = [ "tauri/custom-protocol" ]
+
+[dev-dependencies]
+remove_dir_all = "0.8.0"
```
Comment (sshine) [inline src-tauri/Cargo.toml]: Why is this both a dependency and a dev-dependency?

What does it do to speed up build times?

I don't see it mentioned anywhere.

**Comment by sshine** [comment]
I rebased this branch on top of `main` after merging @Okkonen-GitHub's changes as #18.

Since your commits were signed, they are now no longer verified.

Besides that, the build is a lot quicker, so thanks!

However, the only change that seems to affect this is adding directories to the cache.

Everything else, as far as I see, seems to be left-over changes from experimenting?

File: src-tauri/tauri.conf.json (hunk 1 above)
Comment (ahadley1124) [reply src-tauri/tauri.conf.json:32]: leaving it default and trying to build a production release fails.

File: src-tauri/Cargo.toml (hunk 2 above)
Comment (ahadley1124) [reply src-tauri/Cargo.toml]: `remove_dir_all = "0.8.0"` was mentioned by SNYK as a low security vuln but adding it improved the security rating.

**Comment by ahadley1124** [comment]
Simon, You are correct that the only change in actions was adding more directories to the cache. I did have a couple things from experimenting that I committed for the main branch but nothing that will affect the current running of the build. Please let me know if you have any other questions or comments.

File: src-tauri/Cargo.toml (hunk 2 above)
Comment (sshine) [reply src-tauri/Cargo.toml]: If this dependency is not being used, how about we remove it?

File: src-tauri/tauri.conf.json (hunk 1 above)
Comment (sshine) [reply src-tauri/tauri.conf.json:32]: Gotcha!

> Error You must change the bundle identifier in `tauri.conf.json > tauri > bundle > identifier`. The default value `com.tauri.dev` is not allowed as it must be unique across applications.

Why don't we pick something more canonical:

```suggestion
      "identifier": "chat.simplex.simplex-desktop",
```

Comment (ahadley1124) [reply src-tauri/tauri.conf.json:32]: That identifier should be fine but I would check with the Simplex-Chat owner before publishing a release with it.

File: src-tauri/Cargo.toml (hunk 2 above)
Comment (ahadley1124) [reply src-tauri/Cargo.toml]: That works

**Review by Okkonen-GitHub — APPROVED** [review]
Yeah seems fine

**Review by sshine — APPROVED** [review]
Alright! Sorry for the delays, my internet access is quite sporadic here on vacation.

**Pull request merged**